LowLisp Features.md

```diff
@@ -1,3 +1,39 @@
 # Features Necessary for LowLisp
 
 This document lists everything that will be needed in LowLisp, in order to implement HighLisp.
+
+
+## Language
+
+### Functions
+
+* Are not first class, but experience only a small amount of discrimination. There are function pointers (as in C), and attempting to assign a function to a variable just stores its respective address.
+* Cannot be redefined. However, code can modify a function pointer that will be called in the future. This can be used to implement function redefinition in higher-level languagues.
+* Lack reflection and metaprogramming. Code cannot see which S-expressions make up a function, and cannot modify the function in any way other than overwriting its machine code in memory (probably a bad idea).
+* Must have their parameters and return values defined completely (the generic pointer counts as a defined type).
+
+### Variables
+
+* Are lexically scoped.
+* Are allocated on the stack when declared inside of a function.
+* Are allocated on the heap when declared outside of a function.
+* Must have a type that is known at compile-time, which must make include their size in memory.
+
+### Pointers
+
+* By default are reference-counted, and automatically deallocate their target once the last of their kind falls into oblivion.
+* Are also available in raw form.
+* Must point to a type known at compile time.
```

Comment:
"For raw pointers, void* is a valid type." Sounds like a good idea?

Reply:
Alright. So this is a pointer that does *not* point to a tagged union, but rather to just anything?

Reply:
Yeah, basically, something that the high level compiler/a library handles. Just untyped *something.* (Although I guess a pointer to u8 would have a similar effect)